test/differencer/rulesProcessor.test.ts

```diff
@@ -0,0 +1,129 @@
+    const diffs = [new NodeDiff("test")];
+    const rulesPath = undefined;
+    return expect(applyRules(diffs, rulesPath)).to.eventually.be.rejectedWith(
+      `Invalid rules path: ${rulesPath}`
+    );
+  });
+  it("throws error when the rules file path is null", async () => {
+    const diffs = [new NodeDiff("test")];
+    const rulesPath = null;
+    return expect(applyRules(diffs, rulesPath)).to.eventually.be.rejectedWith(
+      `Invalid rules path: ${rulesPath}`
+    );
+  });
+  it("throws error when the rules file path is empty", async () => {
+    const diffs = [new NodeDiff("test")];
+    const rulesPath = "";
+    return expect(applyRules(diffs, rulesPath)).to.eventually.be.rejectedWith(
+      `Invalid rules path: ${rulesPath}`
+    );
+  });
+  it("throws error when the rules file do not exist", async () => {
+    const diffs = [new NodeDiff("test")];
+    const rulesPath = "/tmp/no-rules.json";
+    return expect(applyRules(diffs, rulesPath)).to.eventually.be.rejectedWith(
+      `Error parsing the rules file '${rulesPath}'`
+    );
+  });
+  it("throws error when the rules file has no valid json", async () => {
+    const tmpFile = tmp.fileSync({ postfix: ".json" });
+    const diffs = [new NodeDiff("test")];
+    return expect(
+      applyRules(diffs, tmpFile.name)
+    ).to.eventually.be.rejectedWith(
+      `Error parsing the rules file '${tmpFile.name}'`
+    );
+  });
+  it("throws error when the rules is not a json array", async () => {
+    const tmpFile = tmp.fileSync({ postfix: ".json" });
+    const rules = {};
+    fs.writeFileSync(tmpFile.name, JSON.stringify(rules));
+    const diffs = [new NodeDiff("test")];
+    return expect(
+      applyRules(diffs, tmpFile.name)
+    ).to.eventually.be.rejectedWith(
+      `Rules must be defined as a json array: ${JSON.stringify(rules, null, 2)}`
+    );
+  });
+  it("exits when rules is an empty array", async () => {
+    const tmpFile = tmp.fileSync({ postfix: ".json" });
+    fs.writeFileSync(tmpFile.name, "[]");
+    const diffs = [new NodeDiff("test")];
+    await applyRules(diffs, tmpFile.name);
+    //verify that diff is not modified
+    expect(diffs).to.deep.equal(diffs);
+    //verify log message
+    sinon.assert.calledWith(loggerSpy, "No rules to apply on the differences");
+  });
+});
+
+describe("Test display name change rule ", () => {
+  const rulesPath = path.join(__dirname, "..", "..", "diffRules", "rules.json");
+  it("applies display name change rule ", async () => {
+    const diff: NodeDiff = {
+      id: "#/web-api/end-points/resource/get",
+      type: ["apiContract:Operation"],
+      added: { "core:name": "newName" },
+      removed: { "core:name": "oldName" }
+    };
+    await applyRules([diff], rulesPath);
+    const diffRule = diff.rule;
+    expect(diffRule.name).to.equal("Rule to detect display name changes");
+    expect(diffRule.type).to.equal("display-name-change");
+    expect(diffRule.params["category"]).to.equal("Breaking");
```

Comment:
The diff rule has a known structure. Use `expect(diffRule).to.deep.equal(expectedStructure)`.

Reply:
see the comment above